docs/development.md

```diff
@@ -0,0 +1,12 @@
+# Developer Details
+
+This documentation stores notes for developers.
+
+## client and api_client packages
```

Comment:
I think moving the "client and api_client packages" section here makes sense since it's more developer-oriented. However, users are still required to update their packages to be compatible with our latest APIs, and I think we should provide instructions or guidance on how to do that. I opened an issue [here](https://github.com/neulab/explainaboard_client/issues/46) for further discussion.